### PR TITLE
Add Rehearsal

### PR DESCRIPTION
**Pull request by oliverlevay**
This adds the "app" Rehearsal.

You can try it out here: https://muntabot.netlify.app/

@bjornregnell will have to update `publish.sh` to match the new upload-to-production flow. 

The new flow is to first run
```bash
sbt fullLinkJS
rm public/main.js
cp target/scala-3.1.0/muntabot-opt/main.js public/main.js

```
and then upload the entire `public` folder to the webserver. This has also been documented in the `README.md`. 

I have also set up some basic CI, that makes sure that the repo actually builds, see an example of how that will look [here](https://github.com/Dsek-LTH/muntabot/actions/runs/1526154354)

### CODE REVIEW

**Comment by bjornregnell** [comment]
cool! 

I'd prefer cp from target to public instead of mv, so the main is still there to run (index-dev.html depends on that).

**Comment by oliverlevay** [comment]
> cool!
> 
> I'd prefer cp from target to public instead of mv, so the main is still there to run (index-dev.html depends on that).

index-dev.html does not depend on `muntabot-opt/main.js` but `muntabot-fastopt/main.js`, but I will change anyway.

**Comment by oliverlevay** [comment]
`--client` does not seem to work on Windows, so I removed it from the instructions.

**Comment by oliverlevay** [comment]
I changed it to an SPA (single-page-application) again, where the `document.location.hash` decides which app to run. No need for multiple html files.

**Comment by bjornregnell** [comment]
Thanks for all you work in this. In the coming days I have not so much time to review the code but I'll try find time to test it before merge. Also I saw that you have created more packages, and (without looking too much into the coupling and cohesion) spontaneously I'd like to keep it simple with one package (or maybe two) without nesting and using top-level defs if relevant and singletons if a new namespace is really needed. Things that are completely independent of the actual app (like addButton) could go into its own module (and we could later make a simple webb-app api similar to introprog for students to use...)   

So basically my aim is: keep structure as simple as possible and use top level definitions in a single module, and only nested modules (packages/ singletons)  if name space is explicitly needed/beneficial.

**Comment by bjornregnell** [comment]
Good if it again can be a single page app :+1: 
Let me know when you feel ready to go live with this and I'll test it locally and publish it.

**Comment by oliverlevay** [comment]
I'm ready to go live!

The package structure right now is 

- shared
- muntabot
- rehearsal

I'm not sure what the best practices are here. `shared.Document` could potentially be converted to a library for students to use

**Comment by oliverlevay** [comment]
> Good if it again can be a single page app :+1: 
> Let me know when you feel ready to go live with this and I'll test it locally and publish it.

I'm ready to go live!

The package structure right now is 

- shared
- muntabot
- rehearsal

I'm not sure what the best practices are here. `shared.Document` could potentially be converted to a library for students to use
